### PR TITLE
[CDAP-13113][MMDS] New experiment/model create flow integration with backend

### PR DESCRIPTION
**Pull request by ajainarayanan**
**TL;DR**:
- Update Experiments/Models create view in UI based on changes in backend
- Changes include,
  - Create new workspace everytime user lands on create experiment or create model states in UI
  - This will involve UI using directives applied on the data from experiment or model.
  - Create split under a model so that model has a reference of split id all the time
  - Polling model status instead of split status as source of truth in split stage.

**Longer description**:
- Previously UI used to store the workspace Id in an experiment to go back to context while creating an experiment or model
- This caused issues while using same file across different experiments or creating multiple models under same experiment
- Since neither experiments nor models can share the workspace we have now moved the workspace information (srcpath & directives) to individual experiment and model.
- Now each time user creates an experiment/model UI goes through the following,

```
  1. Fetch the experiments details
  2. Get directives and srcpath from the details
  3. Create a workspace with the srcpath
  4. Once created execute it with the list of directives
  5. Once the workspace is setup and update the stores.
```
- Since UI has both `srcpath` and `directives` it will create a temporary workspace everytime a user creates a new model in an experiment
- Wrangler has modified its `readFile` api to accept a `scope` parameter which will allow users to create a new workspace for each api call on the same file (meaning isolate different experiments created from the same file).

**Reference diagram from @albertshau on backend states**:
model_lifecycle 1

**Things to improve**:
- Store the file type in experiment/model in addition to directives. This is needed while creating a new workspace
- Right now UI makes execute call twice (1. For updating the newly created workspace to have the directives and 2. When Dataprep loads the workspace to render the table). We should probably have a nice API while creating workspace to accept a list of directives.
- Error handling

JIRA: https://issues.cask.co/browse/CDAP-13113

### CODE REVIEW

File: cdap-ui/app/cdap/components/Experiments/CreateView/index.js [hunk 1]
```diff
@@ -115,6 +114,15 @@ export default class ExperimentCreateView extends Component {
...
   }
   componentWillUnmount() {
+    let {experiments_create} = createExperimentStore.getState();
+    let {workspaceId} = experiments_create;
+    if (workspaceId) {
+      // Every workspace created in experiments create view is temp. So don't worry about deleting it.
+      MyDataPrepApi.delete({
+        namespace: getCurrentNamespace(),
+        workspaceId
+      }).subscribe();
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/Experiments/CreateView/index.js:124]: What's the use of the empty `subscribe`?

File: cdap-ui/app/cdap/components/FileBrowser/index.js [hunk 2]
```diff
@@ -190,14 +190,25 @@ export default class FileBrowser extends Component {
   }
   ingestFile(content) {
     let namespace = NamespaceStore.getState().selectedNamespace;
-
+    let {scope = false} = this.props;
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/FileBrowser/index.js]: nit: can we default `scope: false` under `static defaultProps` instead?

File: cdap-ui/app/cdap/components/FileBrowser/index.js [hunk 3]
```diff
@@ -190,14 +190,25 @@ export default class FileBrowser extends Component {
...
       namespace,
       path: content.path,
       lines: 10000,
       sampler: 'first'
     };
 
+    if (scope) {
+      if (typeof scope === 'string') {
+        params.scope = scope;
+      } else if (typeof scope === 'boolean') {
```
Comment (tonybach) [inline cdap-ui/app/cdap/components/FileBrowser/index.js:213]: Can you help me understand more why `scope` here can be boolean, but also can be string? I saw `scope` being set to `true` in `Experiments/CreateView` component, but then in here we set `scope` to `mmds` if `scope` is true. Why not just set `scope` to `mmds` in the first place?

File: cdap-ui/app/cdap/components/Experiments/CreateView/index.js (hunk 1 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/Experiments/CreateView/index.js:124]: `Rx` won't make a query unless it has a subscription. That is the one of the key difference between a promise and a Observable. So you have to have a dummy subscription even if you don't need the response for rx to make the call for us.

File: cdap-ui/app/cdap/components/FileBrowser/index.js (hunk 3 above)
Comment (ajainarayanan) [reply cdap-ui/app/cdap/components/FileBrowser/index.js:213]: So today wrangler accepts a scope as string (`mmds` or `modelId`) but tomorrow we want to have it as a flag which will accept a boolean value to give us a new workspace id for the same file vs give the same workspace id for the same file. Thats the reason this can be either or. When we add the flag in the wrangler backend it should be simple enough for UI to deprecate it.

**Review by tonybach — COMMENTED** [review]
Code LGTM 👍

**Comment by ajainarayanan** [comment]
Demoed to @bdmogal . Merging.
